### PR TITLE
Split the draft into task binding and task provisioning

### LINKED ISSUE #54: Separate transparency technique from in-line provisioning
https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/500 was closed reasoning that the transparency problem can be solved by taskprov. @cjpatton suggested to split taskprov to enable using the transparency part without in-line provisioning.

### PR DESCRIPTION
**Pull request by cjpatton**
Closes #54.

This commit splits the draft into the task binding extension and a separate section for the optional provisioning mechanism. While at it:

- Fix some editorial issues.
- Rename the extension to "Taskbind".
- Recommend having the author digitally sign the advertised task config. This helps mitigate the risk of malicious Clients DoS'ing an Aggregator.

### CODE REVIEW

File: draft-wang-ppm-dap-taskprov.md [hunk 1]
```diff
@@ -120,28 +118,61 @@ Task configuration:
...
+aborted the request due to not recognizing the task.
+
+Next, the Aggregator encodes the parameters as a `TaskConfig` defined in
+{{task-encoding}} and computes the task ID as above. If the derived task ID
+does not match the task ID of the request, then it MUST reject the report with
+error "invalid_message".
+
+During the upload flow ({{Section 4.4 of !DAP}}), the Leader SHOULD abort the
+request with "unrecognizedTask" if the derived task ID does not match the task
+ID Of the request.
```
Comment (wangshan) [inline draft-wang-ppm-dap-taskprov.md:171]:
```suggestion
ID of the request.
```

File: draft-wang-ppm-dap-taskprov.md [hunk 2]
```diff
@@ -268,22 +304,73 @@ unimplemented DP mechanism).
...
 {{!DAP}}.
 
-## Deriving the Task ID {#construct-task-id}
+# In-band Task Provisioning with the Taskbind Extension {#taskprov}
 
-When using the Taskprov extension, the task ID is computed as follows:
+Before a task can be executed, it is necessary to first provision the Clients,
+Aggregators, and Collector with the task's configuration. The core DAP
+specification does not define a mechanism for provisioning tasks. This section
+describes a mechanism whose  key feature is that task configuration is
```
Comment (wangshan) [inline draft-wang-ppm-dap-taskprov.md:312]: extra space

File: draft-wang-ppm-dap-taskprov.md [hunk 3]
```diff
@@ -268,22 +304,73 @@ unimplemented DP mechanism).
...
+
+Otherwise, if the Leader does opt-in, it advertises the task to the Helper
+during the aggregation protocol ({{Section 4.4 of !DAP}}). In particular, it
+includes the task configuration in an HTTP header of each aggregation job
+request for that task. Before proceeding, the Helper must first parse the
+configuration and decide whether to opt-in; if not, the task's execution halts.
+
+## Task Advertisement
+
+To advertise a task to its peer, a Taskbind participant includes a header
```
Comment (wangshan) [inline draft-wang-ppm-dap-taskprov.md:364]: This should be Taskprov participant?

**Review by wangshan — COMMENTED** [review]
Looks good, only a few minor comments!

File: draft-wang-ppm-dap-taskprov.md (hunk 3 above)
Comment (cjpatton) [reply draft-wang-ppm-dap-taskprov.md:364]: Good catch. This appears to be leftovers from `%s/Taskprov/Taskbind/`. I meant "protocol" here.

**Review by wangshan — APPROVED** [review]

**Comment by cjpatton** [comment]
Rebased and squashed.

**Pull request merged**